apps/src/applab/AppLabView.jsx

```diff
@@ -37,46 +35,31 @@ var AppLabView = React.createClass({
 
   render: function () {
     var playSpaceHeader;
-    if (!this.props.isReadOnlyView) {
+    if (!this.props.isReadOnlyWorkspace) {
       playSpaceHeader = <PlaySpaceHeader
-          hideToggle={this.shouldHideToggle()}
-          hideViewDataButton={this.shouldHideViewDataButton()}
-          startInDesignMode={this.props.startInDesignMode}
-          activeScreenId={this.props.activeScreenId}
+          isEditingProject={this.props.isEditingProject}
           screenIds={this.props.screenIds}
-          onDesignModeButton={this.props.onDesignModeButton}
-          onCodeModeButton={this.props.onCodeModeButton}
           onViewDataButton={this.props.onViewDataButton}
-          onScreenChange={this.props.onScreenChange}
           onScreenCreate={this.props.onScreenCreate} />;
     }
 
     return (
-      <StudioAppWrapper
-          assetUrl={this.props.assetUrl}
-          isEmbedView={this.props.isEmbedView}
-          isShareView={this.props.isShareView}>
+      <ConnectedStudioAppWrapper>
         <div id="visualizationColumn">
           {playSpaceHeader}
           <ProtectedStatefulDiv renderContents={this.props.renderVisualizationColumn} />
         </div>
         <ProtectedStatefulDiv id="visualizationResizeBar" className="fa fa-ellipsis-v" />
-        <ProtectedStatefulDiv
-            id="codeWorkspace"
-            renderContents={this.props.renderCodeWorkspace} />
-      </StudioAppWrapper>
+        <ProtectedStatefulDiv id="codeWorkspace">
+          <ProtectedStatefulDiv id="codeWorkspaceWrapper" renderContents={this.props.renderCodeWorkspace}/>
+          {!this.props.isReadOnlyWorkspace && <ProtectedStatefulDiv id="designWorkspace" style={{display: 'none'}} />}
+        </ProtectedStatefulDiv>
+      </ConnectedStudioAppWrapper>
     );
-  },
-
-  shouldHideToggle: function () {
-    return this.props.isShareView || this.props.isDesignModeHidden;
-  },
-
-  shouldHideViewDataButton: function () {
-    return this.props.isViewDataButtonHidden ||
-        this.props.isDesignModeHidden ||
-        this.props.isShareView ||
-        !this.props.isEditingProject;
   }
 });
-module.exports = AppLabView;
+module.exports = connect(function propsFromState(state) {
```

Comment:
`propsFromStore` sounds good.  Could also leave this anonymous, but it seemed nice to name these.